### ISSUE TITLE
MenuButton doesn't hide the tooltip when used with an animation library

### ISSUE THREAD

**Issue by Dremora** [ariakit/ariakit#2635]
### Current behavior

This occurs when Tooltip is composed with the MenuButton, and Menu is rendered conditionally (to use with Framer Motion). Also, Menu needs to be rendered inside a portal
The example below is based on https://ariakit.org/examples/menu-framer-motion with the added tooltip.

### Steps to reproduce the bug

1. Open sandbox: https://stackblitz.com/edit/bfkems?file=menu-framer-motion%2Fstyle.css,menu-framer-motion%2Fmenu.tsx,menu-framer-motion%2Findex.tsx&theme=dark
2. Hover over "Options" and wait for the tooltip to appear
3. Click the button

### Expected behavior

The tooltip disappears. It currently disappears when Menu is not rendered conditionally, or when portal isn't used.

### Workaround

https://github.com/ariakit/ariakit/issues/2635#issuecomment-1627270013

**Comment by diegohaz** [comment]
Thanks for reporting @Dremora!

Since the portal element is appended to the body after the tooltip is open, it could be a browser extension or even a nested menu inside the tooltip (although unusual, the tooltip component inherits this behavior from hovercard). Maybe we could implement specific checks for tooltips, although I'm still uncertain about the optimal solution in this case.

In your app, you can hide the tooltip when the menu opens:

```jsx
useMenuStore({
  setOpen(open) {
    props.setOpen?.(open);
    if (open) {
      tooltip.hide();
    }
  }
})
```

**Comment by diegohaz** [comment]
This should be fixed in v0.2.15
